ports/nrf/common-hal/neopixel_write/__init__.c

```diff
@@ -138,24 +144,39 @@ void common_hal_neopixel_write (const digitalio_digitalinout_obj_t* digitalinout
         } else {
             uint8_t sd_en = 0;
             (void) sd_softdevice_is_enabled(&sd_en);
-            if (sd_en) {
-                // If the soft device is enabled then we must use PWM to
-                // transmit. This takes a bunch of memory to do so raise an
-                // exception if we can't.
-                pixels_pattern = (uint16_t *) m_malloc(pattern_size, false);
-            } else {
-                pixels_pattern = (uint16_t *) m_malloc_maybe(pattern_size, false);
-            }
 
-            pattern_on_heap = true;
+            if (pixels_pattern_heap_size < pattern_size) {
+                // Current heap buffer is too small.
+                if (MP_STATE_VM(pixels_pattern_heap)) {
+                    // Old pixels_pattern_heap will be gc'd; don't free it.
+                    pixels_pattern = NULL;
+                    MP_STATE_VM(pixels_pattern_heap) = NULL;
+                    pixels_pattern_heap_size = 0;
+                }
```

Comment:
Want to use `realloc` here instead? https://github.com/adafruit/circuitpython/blob/master/py/misc.h#L104 That would reuse the same space if there is room after. (Otherwise it allocates a new chunk.)

Reply:
done